src/Plugin/Hal.php

```diff
@@ -114,6 +114,14 @@ class Hal extends AbstractHelper implements
      */
     protected $urlHelper;
 
+
+    /**
+     * Maximum number of nesting levels
+     *
+     * @var integer
+     */
+    protected $maxDepth;
```

Comment:
I'm thinking this should default to `0`; it makes it more obvious that we need to get back to that value before considering a new object's depth.